pandas/core/reshape/merge.py

```diff
@@ -1730,6 +1730,12 @@ def flip(xs) -> np.ndarray:
         )
         tolerance = self.tolerance
 
+        if is_object_dtype(right_values):
+            raise ValueError("Right input is not from numerical dtype")
+
+        if is_object_dtype(left_values.dtype):
```

Comment:
Should this be loosened to include other non-numeric dtypes for which this should raise?

Reply:
There exist tests for Timedelta and Datetime DataFrames, so this is expected to work I think?. Is there anything you would like to exclude?

Reply:
You might be right, was just curious if this was indeed capturing all the "non-numeric" cases

Reply:
I hope so